### PR TITLE
Add patientLink createdAt index

### PR DESCRIPTION
**Pull request by zdeveloper**
# BACKEND PULL REQUEST

## Related Issue

- partially addresses #4266

## Changes Proposed

- add `PatientLink.createdAt` index to speed up DB query

## Checklist for Primary Reviewer
- [ ] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke tested

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/db/repository/PatientLinkRepository.java [hunk 1]
```diff
@@ -13,7 +13,7 @@ public interface PatientLinkRepository extends EternalAuditedEntityRepository<Pa
 
   @Query(
       value =
-          "select DISTINCT on (ordered.test_order_id) * from (select * from {h-schema}patient_link where test_order_id in :testOrderIds order by created_at desc ) ordered where test_order_id in :testOrderIds",
+          "select DISTINCT on (ordered.test_order_id) * from (select * from {h-schema}patient_link where test_order_id in :testOrderIds order by created_at desc ) ordered",
```
Comment (zdeveloper) [inline backend/src/main/java/gov/cdc/usds/simplereport/db/repository/PatientLinkRepository.java:16]: removing extra where clause per @mehansen feedback, discussed multiple implementations and sub-querys tended to perform the best

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml [hunk 2]
```diff
@@ -4115,4 +4115,20 @@ databaseChangeLog:
...
+        - createIndex:
+            tableName: patient_link
+            indexName: ix__patient_link__created_at
+            columns:
+              - column:
+                  name: created_at
+                  descending:  true
+      rollback:
+        - dropIndex:
+            indexName: ix__patient_link__created_at
```
Comment (zdeveloper) [inline backend/src/main/resources/db/changelog/db.changelog-master.yaml:4134]: Adding an index to this will make the [`PatientLinkRepository.findMostRecentByTestOrderIdIn`](https://github.com/CDCgov/prime-simplereport/pull/4293/files#r975407868) query a lot faster since it orders by `created_at`

File: backend/src/main/java/gov/cdc/usds/simplereport/db/repository/PatientLinkRepository.java (hunk 1 above)
Comment (mehansen) [reply backend/src/main/java/gov/cdc/usds/simplereport/db/repository/PatientLinkRepository.java:16]: when we removed the outer where, didn't it make it just as slow as removing the subquery though? I thought no outer where and no subquery ended up being about the same for some weird reason

Comment (zdeveloper) [reply backend/src/main/java/gov/cdc/usds/simplereport/db/repository/PatientLinkRepository.java:16]: yes it did, but not after I added the index!

File: .github/workflows/testDBChangelog.yml [hunk 3]
```diff
@@ -41,5 +41,5 @@ jobs:
         run: touch ../.env && docker compose -f ../docker-compose.yml up -d db
       - name: Run forward migrations
         run: ./gradlew liquibaseUpdate
-      - name: Test rolling back all migrations
-        run: ./gradlew liquibaseRollbackCount -PliquibaseCommandValue=9999
+      - name: Test rolling back on last 20 migrations
+        run: ./gradlew liquibaseRollbackCount -PliquibaseCommandValue=20
```
Comment (zdeveloper) [inline .github/workflows/testDBChangelog.yml:45]: See why this change https://github.com/CDCgov/prime-simplereport/wiki/Database-maintenance#rollback-gotchas

**Review by emmastephenson — APPROVED** [review]

File: backend/src/main/java/gov/cdc/usds/simplereport/db/repository/PatientLinkRepository.java (hunk 1 above)
Comment (zdeveloper) [reply backend/src/main/java/gov/cdc/usds/simplereport/db/repository/PatientLinkRepository.java]: This has been moved to here https://github.com/CDCgov/prime-simplereport/pull/4319 per @emmastephenson feedback

**Review by alismx — APPROVED** [review]
LGTM, thanks for adding that note in the wiki

**Review by DanielSass — APPROVED** [review]
